### ISSUE TITLE
Menus of panels don't close when losing (or don't lose) focus

### ISSUE THREAD

**Issue by Saviq** [MirServer/mir#2585]
Open a drop-down menu of mate-panel, then focus a window. You can see the panel menu staying on.
![frame_2022-08-25T11:31+02:00](https://user-images.githubusercontent.com/231077/186629998-fa14f81b-ba70-4169-943f-7b6bc0a2f855.png)

Using this [Miriway](https://github.com/Miriway/Miriway) config:

```ini
x11-window-title=Miriway
idle-timeout=180
ctrl-alt=t:miriway-terminal # Default "terminal emulator finder"

enable-x11=true

shell-component=swaybg -i /usr/share/backgrounds/warty-final-ubuntu.png
shell-component=mate-panel
meta=a:mate-panel --run-dialog
ctrl-alt=t:mate-terminal
```

**Comment by wmww** [comment]
The issue is the the current implementation dismisses popups when they lose focus, however panels generally don't gain or loose focus. This can be tested by running a recent version of gtk-layer-demo. When keyboard is set to none (never focused) or exclusive (always focused), popups don't close when other windows are clicked. When it's set to on-demand they do.

**Comment by Saviq** [comment]
That also means you can't navigate the menus with the keyboard, which would be another issue to fix?

**Comment by wmww** [comment]
The various keyboard interactivity modes are defined in the layer shell protocol and AFAIK we correctly implement them. If a change in behavior there was desirable that would need to be patched in mate-panel.

**Comment by wmww** [comment]
In my investigation I discovered #2604, which will likely be fixed at the same time

**Comment by Saviq** [comment]
Fixed by #2606
